peps/pep-0733.rst

```diff
@@ -0,0 +1,660 @@
+`SWIG <https://swig.org/>`__ for C/C++, or
+`Python.NET <https://github.com/pythonnet/pythonnet>`__ for .NET (C#).
+`HPy <https://hpyproject.org>`__ and
+`pythoncapi-compat <https://pythoncapi-compat.readthedocs.io/en/latest/>`__.
+CPython's DSL for parsing function arguments, the
+`Argument Clinic <https://devguide.python.org/development-tools/clinic/>`__,
+can also be seen as belonging to this category of stakeholders.
+
+Alternative APIs need minimal building blocks for accessing CPython
+efficiently. They don't necessarily need an ergonomic API, because
+they typically generate code that is not intended to be read
+by humans. But they do need it to be comprehensive enough so that
+they can avoid accessing internals, without sacrificing performance.
+
+Binding generators often need to:
+
+* Create custom objects (e.g. function/module objects
+  and traceback entries) that match the behavior of equivalent
+  Python code as closely as possible.
+* Dynamically create objects which are static in traditional
+  C extensions (e.g. classes/modules), and need CPython to manage
+  their state and lifetime.
+* Dynamically adapt foreign objects (strings, GC'd containers), with
+  low overhead.
+* Adapt external mechanisms, execution models and guarantees to the
+  Python way (green threads/continuations, one-writer-or-multiple-readers
+  semantics, virtual multiple inheritance, 1-based indexing, super-long
+  inheritance chains, goroutines, channels, etc.).
+
+These tools might also benefit from a choice between a more stable
+and a more performant (possibly lower-level) API. Their users could
+then decide whether they can afford to regenerate the code often or
+trade some performance for more stability and less maintenance work.
+
+
+Strengths of the C API
+======================
+
+While the bulk of this document is devoted to problems with the
+C API that we would like to see fixed in any new design, it is
+also important to point out the strengths of the C API, and to
+make sure that they are preserved.
+
+As mentioned in the introduction, the C API enabled the
+development and growth of the Python ecosystem over the last
+three decades, while evolving to support use cases that it was
+not originally designed for. This track record in itself is
+indication of how effective and valuable it has been.
+
+A number of specific strengths were mentioned in the
+capi-workgroup discussions. Heap types were identified
+as much safer and easier to use than static types
+[`Issue 4 <https://github.com/capi-workgroup/problems/issues/4#issuecomment-1542324451>`__].
+
+API functions that take a C string literal for lookups based
+on a Python string are very convenient
+[`Issue 30 <https://github.com/capi-workgroup/problems/issues/30#issuecomment-1550098113>`__].
+
+The Limited API and stable ABI hide implementation details and
+make it easier to evolve Python
+[`Issue 30 <https://github.com/capi-workgroup/problems/issues/30#issuecomment-1560083258>`__].
```

Comment:
Any API that hides implementation details makes it easier to evolve the implementation. It doesn't have to be limited.
If the point here is that a limited API helps evolution, I disagree.
An API that is not tied to implementation details is what is important. It does have to be "limited".

So maybe change this to something like:
"The limited API shows that an API that hide implementation details makes it easier to evolve Python"
The stability of the ABI and the extent of the API are also largely unrelated.